File: 3375-minimum-operations-to-make-array-values-equal-to-k/3375-minimum-operations-to-make-array-values-equal-to-k.cs
public class Solution {
    public int MinOperations(int[] nums, int k) {
        var uniq = new HashSet<int>();
        foreach(var num in nums){
            if(num < k) return -1;              // not possible to convert it to k
            else if(num > k) uniq.Add(num);     // all bigger nums need to be converted
        }

        return uniq.Count;
    }
}

File: 3375-minimum-operations-to-make-array-values-equal-to-k/3375-minimum-operations-to-make-array-values-equal-to-k.java
class Solution {
    public int minOperations(int[] nums, int k) {
        var uniq = new HashSet();
        for(var num : nums){
            if(num < k) return -1;              // not possible to convert it to k
            else if(num > k) uniq.add(num);     // all bigger nums need to be converted
        }

        return uniq.size();
    }
}